### PR TITLE
adding linear probing

### PR DESCRIPTION
**Pull request by aaadddiii**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

fixes issue #3898 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

issue #3898

### CODE REVIEW

File: C/ds/Linear_probing.c [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+// 9 0
+// Enter your choice
+// 1.insert
+// 2.Does Exist
+// 3.Print Hashtable
+// 4.Exit
+// 4
+
+// Time complexity : O(n)
+// Space complexity : O(1)
```
Comment (HarshCasper) [inline C/ds/Linear_probing.c:166]: Put in a multi-line comment

File: C/ds/Linear_probing.c [hunk 2]
```diff
@@ -0,0 +1,166 @@
+//Linear probing is a collision resolution technique in hashtables
+//In linear probing, we linearly probe for next empty slot in the hashtable for inserting an element
```
Comment (HarshCasper) [inline C/ds/Linear_probing.c:2]: Add a better description

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
@aaadddiii Found two PR for the same issue #3951 and #4088. Closing this as of now. Open only one PR. It becomes difficult for us which PR to review.

**Comment by aaadddiii** [comment]
one is an implementation in c and other in c++

**Pull request closed without merging**